local-queue.js

```diff
@@ -0,0 +1,73 @@
+const { weightSrvRecords } = require("ioredis/built/cluster/util")
+
+const queues = {}
+
+// Get or create queue
+function getOrCreateQueue(queueName) {
+  q = null
+  if (!(queueName in queues)) {
+    q = []
+    queues[queueName] = q 
+  } else {
+    q = queues[queueName]
+  }
+  return q
+}
+
+// Enqueue an item to the queue
+async function push(queueName, item) {
+  q = getOrCreateQueue(queueName)
+  // Check if the element already exists in the list
+  if (!(item in q)) {
```

Comment:
This doesn't work, run e.g.

```javascript
let q = [4,5,6];
4 in q; //false
q.includes(4); //true
```

Furthermore, for checking if a data structure contains an item, a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) is faster when the data structure is large. What kind of sizes do you expect `q` to be?

Reply:
queue should be a few hundred items. Sets indeed are faster but not sure if sets have guaranteed order of adding hence using the array push and pop

Reply:
They do, see the [description](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#description), though at a few hundred items, I think it's fine for now. If we ever have performance issues, we should take this into consideration.